ragen/llm_agent/es_manager.py

```
@@ -66,18 +111,118 @@ def _init_envs(self):
+            if isinstance(base_search_config, dict):
+                base_config_dict = copy.deepcopy(base_search_config)
+            elif isinstance(base_search_config, DictConfig):
+                base_config_dict = copy.deepcopy(OmegaConf.to_container(base_search_config, resolve=True))
+            elif is_dataclass(base_search_config):
+                base_config_dict = copy.deepcopy(asdict(base_search_config))
+            else:
+                # Fallback: try to convert using vars() or dict()
+                try:
+                    base_config_dict = copy.deepcopy(dict(base_search_config) if hasattr(base_search_config, '__iter__') else vars(base_search_config))
+                except (TypeError, ValueError) as e:
+                    raise TypeError(f"base_search_config must be a dict, OmegaConf DictConfig, or dataclass instance, got {type(base_search_config)}: {e}")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)